### PR TITLE
Add documentation on how to enable bare-metal

### PR DESCRIPTION
**Pull request by aashishc1988**
Add documentation that guides the user on how to enable bare-metal

### CODE REVIEW

**Comment by theotherjimmy** [comment]
Requires https://github.com/armmbed/mbed-os/pull/9800

**Review by theotherjimmy — APPROVED** [review]

**Review by SenRamakri — APPROVED** [review]

**Review by AnotherButler — APPROVED** [review]
Thanks for the PR 👍

**Review by bulislaw — COMMENTED** [review]
I think we should introduce this not only as a migration option, but add a section describing differences between the two. Here we could add the mode description/differences https://os.mbed.com/docs/mbed-os/v5.11/reference/mbed-os-fundamentals.html and link it to here https://os.mbed.com/docs/mbed-os/v5.11/tools/working-with-mbed-cli.html after we add a section how to use it.

**Comment by SenRamakri** [comment]
@bulislaw - A high-level introduction/marketing content about bare-metal is in works from Mohit. He will be working with Amanda to bring that as a PR. It will also capture the differences from MbedOS5.

**Comment by AnotherButler** [comment]
@bulislaw Have your concerns been addressed (at least until we receive the introduction and comparison)?

File: docs/tutorials/migrate-from-mbed-os-2-to-5.md [hunk 1]
```diff
@@ -348,3 +348,13 @@ Now, the program successfully compiles.
 ### Runtime errors
 
 Although the program or library now compiles successfully, runtime errors may still be present. Please visit the [compile-time errors tutorial](compile-time-errors.html#runtime-errors-and-lights-of-the-dead) for further debugging tips about common errors.
+
+### Enabling Mbed OS bare metal
+
+Enabling the Mbed OS bare metal profile allows you to build Mbed OS without an RTOS. To enable it, you have to complete the [migration to Mbed OS 5](#migrating-to-mbed-os-5). Once the migration is complete, you can enable Mbed OS bare metal by creating an `mbed-app.json` with the following contents:
```
Comment (thegecko) [inline docs/tutorials/migrate-from-mbed-os-2-to-5.md:354]: @AnotherButler isn't this file called `mbed_app.json`? (with an underscore)

Comment (AnotherButler) [reply docs/tutorials/migrate-from-mbed-os-2-to-5.md:354]: I'll fix that

Comment (thegecko) [reply docs/tutorials/migrate-from-mbed-os-2-to-5.md:354]: hooray!

**Pull request merged**